### ISSUE TITLE
Allows submitting invalid queries

### ISSUE THREAD

**Issue by runeh** [graphql/graphiql#2721]
graphiql2 does not prevent users from trying to submit malformed queries.  This is different from 1.x graphqiql versions

## Steps to reproduce:

1. go to [the demo app](https://graphiql-test.netlify.app/?query=%23%20Welcome%20to%20GraphiQL%0A%23%0A%23%20GraphiQL%20is%20an%20in-browser%20tool%20for%20writing%2C%20validating%2C%20and%0A%23%20testing%20GraphQL%20queries.%0A%23%0A%23%20Type%20queries%20into%20this%20side%20of%20the%20screen%2C%20and%20you%20will%20see%20intelligent%0A%23%20typeaheads%20aware%20of%20the%20current%20GraphQL%20type%20schema%20and%20live%20syntax%20and%0A%23%20validation%20errors%20highlighted%20within%20the%20text.%0A%23%0A%23%20GraphQL%20queries%20typically%20start%20with%20a%20%22%7B%22%20character.%20Lines%20that%20start%0A%23%20with%20a%20%23%20are%20ignored.%0A%23%0A%23%20An%20example%20GraphQL%20query%20might%20look%20like%3A%0A%23%0A%23%20%20%20%20%20%7B%0A%23%20%20%20%20%20%20%20field(arg%3A%20%22value%22)%20%7B%0A%23%20%20%20%20%20%20%20%20%20subField%0A%23%20%20%20%20%20%20%20%7D%0A%23%20%20%20%20%20%7D%0A%23%0A%23%20Keyboard%20shortcuts%3A%0A%23%0A%23%20%20%20Prettify%20query%3A%20%20Shift-Ctrl-P%20(or%20press%20the%20prettify%20button)%0A%23%0A%23%20%20Merge%20fragments%3A%20%20Shift-Ctrl-M%20(or%20press%20the%20merge%20button)%0A%23%0A%23%20%20%20%20%20%20%20%20Run%20Query%3A%20%20Ctrl-Enter%20(or%20press%20the%20play%20button)%0A%23%0A%23%20%20%20%20Auto%20Complete%3A%20%20Ctrl-Space%20(or%20just%20start%20typing)%0A%23%0A%0A). There is no query, just comments in the edit view.
2. Click the "Execute query" button

## Expected result:

Nothing happens. Or some kind of "malformed query" message?

## Actual result:

graphiql tries to parse and submit the query, causing the following error to be printed in the result pane:

```
{
  "errors": [
    {
      "message": "Invalid AST Node: undefined.",
      "stack": "Error: Invalid AST Node: undefined.\n    at o (https://graphiql-test.netlify.app/graphiql.min.js:1:195952)\n    at Le (https://graphiql-test.netlify.app/graphiql.min.js:1:233026)\n    at e.isSubscriptionWithName (https://graphiql-test.netlify.app/graphiql.min.js:29:466877)\n    at https://graphiql-test.netlify.app/graphiql.min.js:29:464592\n    at https://graphiql-test.netlify.app/graphiql.min.js:29:223596\n    at https://graphiql-test.netlify.app/graphiql.min.js:29:270611\n    at t.<computed> (https://graphiql-test.netlify.app/graphiql.min.js:29:262405)\n    at doHandleBinding (https://graphiql-test.netlify.app/graphiql.min.js:1:134678)\n    at https://graphiql-test.netlify.app/graphiql.min.js:1:135710\n    at lookupKey (https://graphiql-test.netlify.app/graphiql.min.js:1:124311)"
    }
  ]
}
```

Maybe it makes sense that there's an error rather than doing nothing, but seems like it make sense to at least ignore empty queries.

**Comment by thomasheyenbrock** [comment]
Thanks for reporting this. Just to clarify, the intended behavior is that a HTTP request is triggered for _both_ valid and invalid queries. (That also is the behavior of `graphiql@1.x`.) The final validation of the query should always happen on the server.

Currently (`graphiql@2.0.2`), no network request is dispatched at all.

**Comment by vitalikda** [comment]
@thomasheyenbrock thank you for your reply! 

However, if you do want to prevent submitting an empty request on client side, how would you recommend handling it? 

In my case, I replace fetcher with custom error message, which does not seem ideal. I went with this approach due to couldn't figure out how to disable hot key `Ctrl+Enter` for inexecutable queries.

**Comment by thomasheyenbrock** [comment]
Hey @vitalikda 👋 the solution you describe is currently the only supported one. The fetcher function will always be called, there's no way yet to opt-out of this. However, I do think this is a problem that we should explore deeper and identify possible solutions. Would you mind opening a new issue for this? (That we can close this once the PR to fix this is in and have a better separation between the bug and your feature request.)

**Comment by thomasheyenbrock** [comment]
Just released the fix for this in `graphiql@2.0.4` and `@graphiql/toolkit@0.7.1`
